src/cli/controller.js

```diff
@@ -106,14 +106,17 @@ class Controller {
     if (!fs.existsSync(opts.source)) {
       throw new Error('Provided directory does not exist.');
     }
-
+    
     return this.client.generateUploadUrl(this.config)
       .then(([body]) => {
         cloudfunction.sourceUploadUrl = body.uploadUrl;
         CloudFunction.addLocaldir(cloudfunction, opts.source);
 
         return new Promise((resolve, reject) => {
           // Parse the user's code to find the names of the exported functions
+          if (opts.firebase) {
+            resolve(name.replace(/\-/g, '.'));
```

Comment:
I see you're resolving the promise right here, do you also want to add a `return` statement to avoid the `exec` call on line 120?